enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,404 @@
+those fields already.
+
+Validation of CredentialsRequest by this new token-aware CCO? Maybe. Currently, CCO gets some permissions under ROSA
+(unused), could expand these permissions to include tools like AWS's Policy Simulator such that it could validate a role
+has permissions, and with this CCO could be the alerting mechanism for a changed CredentialsRequest without sufficient 
+permissions.
+
+CCO should consider adding the above, three-part test as a utility function exposed in the API.
+
+**HyperShift changes**: Include Cloud Credential Operator with token-aware mode (see above). Allows for processing of 
+CredentialsRequest objects added by operators.
+
+**OperatorHub and Console changes**: Allow for input from user of additional fields during install depending on the 
+Cloud Provider which will result in ENV variables on the Subscription object that are required in the 
+CredentialsRequests created by the Operator. Setting the Subscription config.ENV will allow UX for the information 
+needed by CCO or webhook, for input of the cloud credentials while not having to change the Subscription API.
+CLI Users will need to add the following to their subscription:
+
+```yaml
+kind: Subscription
+metadata:
+ name: ...
+spec:
+  config:
+    env:
+    - name: ROLEARN
+      value: "<role ARN >"
+```
+
+Show in OperatorHub that the cluster is in a mode that supports token-based authentication by reading the 
+`.spec.serviceAccountIssuer` from the Authentication CR, `.status.platformStatus.type` from the Infrastructure CR,
+`.spec.credentialsMode` from the CloudCredentials CR.
+
+Show that the operator is enabled for Token-based use by reading the CSV annotation provided by the operator author.
+
+Generate a manual ack for operator upgrade when cloud resources are involved. This can be determined by parsing the 
+Subscription objects. Ensures admin signs off that cloud resources are provisioned and working as intended before an 
+operator upgrade. This could be by setting the default subscription mode to Manual in the UI for operators that support 
+TAT.
+
+**Operator team changes**: Follow new guidelines for allowing for the operator to work on token auth 
+enabled cluster. New guidelines would include the following to use CCO has detected cluster is using time-based tokens:
+
+- Changing CSV to add CredentialRequest RBAC
+- Add CredentialRequests into a defined directory in the bundle
+- Add a bundle annotation claiming token-based authentication support
+- Add a script in the operator description to help set up the correct role
+- (Optional) Add the projected ServiceAccount volume to the Deployment embedded in the CSV;
+- handle the Secret themselves (like today’s CCO Mint mode, read and use the credentials in the Secret whose name they 
+  know from the CredentialsRequest and is in the same Namespace)
+- Add structure to controller code to alert on missing cloud credentials, not crash.
```

Comment:
i think this section may be outdated relative to what it is in https://docs.google.com/document/d/1iFNpyycby_rOY1wUew-yl3uPWlE00krTgr9XHDZOTNo/edit